Sources_v3/StreamChat/WebSocketClient/Events/ReactionEvents.swift

```diff
@@ -12,15 +12,17 @@ public struct ReactionNewEvent<ExtraData: ExtraDataTypes>: EventWithReactionPayl
     public let reactionScore: Int
     public let createdAt: Date
     
+    let reaction: MessageReactionPayload<ExtraData>
     let payload: Any
     
     init(from response: EventPayload<ExtraData>) throws {
-        userId = try response.value(at: \.user?.id)
+        userId = try response.value(at: \.reaction?.user.id)
         cid = try response.value(at: \.cid)
-        messageId = try response.value(at: \.message?.id)
+        messageId = try response.value(at: \.reaction?.messageId)
         reactionType = try response.value(at: \.reaction?.type)
         reactionScore = try response.value(at: \.reaction?.score)
         createdAt = try response.value(at: \.reaction?.createdAt)
+        reaction = try response.value(at: \.reaction)
         payload = response
     }
 }
```

Comment:
This is not really what want, take a look at `MemberEvent` and `TypingEvent` for example. The fact, that events are generic is wrong and leftover from a wrong implementation. Only the initializer should be generic. 

The payload itself is stored in the type-erased form in `payload`.

Reply:
fixed in a788b191cbddc1055753012de3cc2b8945d34c9f